nomad/structs/structs.go

```diff
@@ -3676,6 +3693,11 @@ func (a *AllocatedMemoryResources) Add(delta *AllocatedMemoryResources) {
 	}
 
 	a.MemoryMB += delta.MemoryMB
+	if delta.MemoryMaxMB != 0 {
+		a.MemoryMaxMB += delta.MemoryMaxMB
+	} else {
+		a.MemoryMaxMB += delta.MemoryMB
+	}
```

Comment:
The updates in AllocatedMemoryResource tracking isn't strictly needed. I'm adding them for consistency and to ease having the scheduler consider MemoryMaxMB in the future.